### PR TITLE
feat: rate limit insight update consumer

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

When running to an unconstrained rate limit with a high rate of producer enqueuing this task we reached 65 executions per minute across 5 workers before ClickHouse queries started to slow down.

By setting a rate limit on how fast we process queued cache updates we can keep the queue of tasks full so that workers aren't idle without overwhelming ClickHouse

When #10991 lands controlling the rate at which we process updates will be even more important

## Changes

Starts with no rate limit, on workers start sets rate limit from instance settings, and listens to changes in instance settings to update the rate limit

## How did you test this code?

running it locally and seeing processing change speed

### CODE REVIEW

File: posthog/celery.py [hunk 1]
```diff
@@ -428,15 +429,31 @@ def check_cached_items():
...
+    from structlog import get_logger
+
+    logger = get_logger(__name__)
+
+    if rate_limit is not None:
+        logger.info("set_update_cache_consumer_rate_limit", rate_limit=rate_limit)
+        app.control.rate_limit("posthog.celery.update_cache_item_task", rate_limit)
+
+
+@app.task(ignore_result=False, rate_limit="6/m")
```
Comment (benjackwhite) [inline posthog/celery.py:443]: The default is `6/m` - is that not too slow? Or am I reading the value wrong?

File: posthog/celery.py [hunk 2]
```diff
@@ -428,15 +429,31 @@ def check_cached_items():
...
-def update_cache_item_task(key: str, cache_type, payload: dict) -> None:
+@app.task(ignore_result=True)
+def update_cache_consumer_rate_limit(rate_limit: Optional[Union[str, int]]) -> None:
+    from structlog import get_logger
+
+    logger = get_logger(__name__)
+
+    if rate_limit is not None:
+        logger.info("set_update_cache_consumer_rate_limit", rate_limit=rate_limit)
+        app.control.rate_limit("posthog.celery.update_cache_item_task", rate_limit)
```
Comment (benjackwhite) [inline posthog/celery.py:440]: This takes care of updates, what about on initialisation? If I'm reading these changes right, on load we will use the default set on line 443 of "6/m" and then only if the instance setting is updated will we change it to something else. Feels like we need to read the instance setting on startup?

**Review by benjackwhite — CHANGES_REQUESTED** [review]

File: posthog/celery.py (hunk 2 above)
Comment (pauldambra) [reply posthog/celery.py:443]: Ah! Much better 💖

File: posthog/celery.py (hunk 1 above)
Comment (pauldambra) [reply posthog/celery.py]: No, that's right... I took a guess at "not so fast that it'd be a problem" but init'g on start is way nicer

**Review by benjackwhite — APPROVED** [review]

**Pull request merged**